### DISCUSSION TITLE
DataTable Sort Toggle & State

### DISCUSSION THREAD

**Discussion by aaronst** [Textualize/textual#3301] [Ideas]
As a user, I often want to be able to click a column header to sort it. For subsequent clicks, I want the sort order to be reversed. It would be great to not have to manage the order for each column myself, and instead have this be managed internally by the `DataTable`. Being able to know what the current "sort state" is for a `DataTable` would be very helpful.

**Comment by TomJGooding** [comment]
I like the idea of this convenience with a DataTable, but I'm not sure how it might work for sorting that is more complex (see #3090).

**Reply by aaronst** [reply]
Awesome, I missed that PR. I was thinking there would be something like the `current_sorts` in that PR, but using a data structure that also maintains the order of the columns. With the addition of custom sort keys though, I agree that there may be more consideration needed.
